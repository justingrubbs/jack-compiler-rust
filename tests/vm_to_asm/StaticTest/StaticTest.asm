@256
D=A
@SP
M=D
@StaticTest.Sys.init$ret.1
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Sys.init
0;JMP
(StaticTest.Sys.init$ret.1)
@111
D=A
@SP
A=M
M=D
@SP
M=M+1
@333
D=A
@SP
A=M
M=D
@SP
M=M+1
@888
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
@StaticTest.8
M=D
@SP
AM=M-1
D=M
@StaticTest.3
M=D
@SP
AM=M-1
D=M
@StaticTest.1
M=D
@StaticTest.3
D=M
@SP
A=M
M=D
@SP
M=M+1
@StaticTest.1
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
M=M-D
@StaticTest.8
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
A=A-1
M=D+M
