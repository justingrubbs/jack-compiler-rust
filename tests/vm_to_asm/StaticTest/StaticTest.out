|RAM[256]|
|     51 |
